test/test_model.py

```diff
@@ -167,3 +174,38 @@ def test_config(self):
         with self.assertRaises(TypeError):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
+
+
+def fake_script(test_case, name, content):
+    if not hasattr(test_case, 'fake_script_path'):
+        fake_script_path = tempfile.mkdtemp('-fake_script')
+        os.environ['PATH'] = f'{fake_script_path}:{os.environ["PATH"]}'
+
+        def cleanup():
+            shutil.rmtree(fake_script_path)
+            os.environ['PATH'] = os.environ['PATH'].replace(fake_script_path + ':', '')
+
+        test_case.addCleanup(cleanup)
+        test_case.fake_script_path = pathlib.Path(fake_script_path)
+
+    with open(test_case.fake_script_path / name, "w") as f:
+        # Before executing the provided script, dump the provided arguments in calls.txt.
+        f.write('#!/bin/bash\n{ echo -n $(basename $0); for s in "$@"; do echo -n \;$s; done; echo; } >> $(dirname $0)/calls.txt\n' + content)
```

Comment:
Two backward slashes are needed I believe:

```\\;$s```

Having just one raises ```W605 invalid escape sequence '\;'```

Python corrects it automatically though but this will change.

https://docs.python.org/3/whatsnew/3.6.html#deprecated-python-behavior

> A backslash-character pair that is not a valid escape sequence now generates a DeprecationWarning. Although this will eventually become a SyntaxError, that will not be for several Python releases.

Reply:
Aha, thanks, I didn't know that was now deprecated.